arangod/Aql/ExecutionNode.cpp

```diff
@@ -2824,14 +2824,14 @@ ExecutionNode* AsyncNode::clone(ExecutionPlan* plan, bool withDependencies,
 }
 
 namespace {
-const char* MATERIALIZE_NODE_IN_NM_COL_PARAM = "inNmColPtr";
-const char* MATERIALIZE_NODE_IN_NM_DOC_PARAM = "inNmDocId";
-const char* MATERIALIZE_NODE_OUT_VARIABLE_PARAM = "outVariable";
+constexpr std::string_view MATERIALIZE_NODE_IN_NM_DOC_PARAM = "inNmDocId";
+constexpr std::string_view MATERIALIZE_NODE_OUT_VARIABLE_PARAM = "outVariable";
+constexpr std::string_view MATERIALIZE_NODE_MULTI_NODE_PARAM = "multiNode";
 }  // namespace
 
 MaterializeNode* materialize::createMaterializeNode(
-    ExecutionPlan* plan, arangodb::velocypack::Slice const& base) {
-  if (base.hasKey(MATERIALIZE_NODE_IN_NM_COL_PARAM)) {
+    ExecutionPlan* plan, arangodb::velocypack::Slice const base) {
```

Comment:
If this code runs on DB servers during upgrade, the following may happen:
* the coordinator is running version 3.10 and sends AQL snippet data containing the "inNmColPtr" key in order to create a MaterializeMultiNode.
* the DB server is running version 3.11 already and does not interpret the `inNmColPtr` key anymore, and thus creates a MaterializeSingleNode.

Reply:
This is handled  in arangod/Aql/IResearchViewNode.cpp:1124   - We will fail View enumeration node creation.